lib/mojito.js

```diff
@@ -130,7 +130,17 @@ Mojito = (function ()
             // Call trigger
             this.activate = this.activate.bind(this);
             this.trackExposureEvent = this.trackExposureEvent.bind(this);
-            this.options.trigger(this);
+
+            try
+            {
+                this.options.trigger(this);
+            }
+            catch (e)
+            {
+                this.chosenRecipe = {};
+                this.trackRecipeFailureEvent(e);
+                this.log("Test Object [" + this.options.name + "][" + this.options.id + "] trigger failed, error: " + (e.message||e), 'error');
+            }
```

Comment:
Perfect! Would you mind issuing this as a separate PR so we can keep the commits as tight as possible?

Also, why is `chosenRecipe` here set with an empty object? Is that a protection for an issue later in the library code? Or are you looking at handling it for this check in our usual storage Adapters?:

```js
        // Refresh the page unless we're in a trigger or preview mode
        var preview = document.location.search.indexOf('mojito_' + obj.options.id + '=' + obj.chosenRecipe.id) > -1;
        if (obj.chosenRecipe.name && !obj.options.divertTo && !preview)  // Here
        {
```

Source: https://mojito.mx/docs/js-delivery-api-storage-adapter#mojitooptionsstorageadapteronrecipefailure

Reply:
@dapperdrop 
I have removed this change from this PR and will send another PR once this is merged. And you're right, the reason I set ```chosenRecipe``` with an empty object is: we don't have a trigger failure event yet so I use RecipeFailureEvent instead. In this case ```chosenRecipe``` is undefined, codes that referring it might throw a undefined error. e.g. ```obj.chosenRecipe.name```.